third_party/python/pip.bzl

```diff
@@ -2,152 +2,12 @@
-        args,
-        environment = {
-            "SOURCE_DATE_EPOCH": "315532800",
-            "PYTHONHASHSEED": "0",
-        },
-        timeout = rctx.attr.timeout,
-        quiet = rctx.attr.quiet,
-    )
-    if result.return_code:
-        fail("pip_wheel failed: {} ({})".format(result.stdout, result.stderr))
-
-    # Next, we download/build all the Python wheels for each requirement.
-    args = [
-        rctx.path(rctx.attr.python_interpreter),
-        "-m",
-        "pip",
-        "wheel",
-        "--use-pep517",
-        "-r",
-        rctx.path(rctx.attr.requirements),
-        "-w",
-        "./",
-    ]
-    rctx.report_progress("Pre-building Python wheels")
-    result = rctx.execute(
-        args,
-        environment = {
-            "SOURCE_DATE_EPOCH": "315532800",
-            "PYTHONHASHSEED": "0",
-        },
-        timeout = rctx.attr.timeout,
-        quiet = rctx.attr.quiet,
-    )
-    if result.return_code:
-        fail("pip_wheel failed: {} ({})".format(result.stdout, result.stderr))
-
-    # Last, we generate a new Python requirements file that does not contain any
-    # VCS links. We avoid just patching the existing python_requirements.txt
-    # file, and instead generate a requirements file based on the wheels that
-    # were built above. This enables us to make changes to the main
-    # python_requirements.txt file without impacting this step.
-    args = [rctx.path(rctx.attr._gen_requirements_script)]
-    rctx.report_progress("Generating sanitzed requirements file")
-    result = rctx.execute(
-        args,
-        environment = {
-            "SOURCE_DATE_EPOCH": "315532800",
-            "PYTHONHASHSEED": "0",
-        },
-        timeout = rctx.attr.timeout,
-        quiet = rctx.attr.quiet,
-        working_directory = "./",
-    )
-    if result.return_code:
-        fail("pip_wheel failed: {} ({})".format(result.stdout, result.stderr))
-
-    # We need a BUILD file to load the downloaded Python packages.
-    rctx.file(
-        "BUILD.bazel",
-        _WHEEL_BUILD_FILE_CONTENTS,
-    )
-
-pip_wheel = repository_rule(
-    implementation = _pip_wheel_impl,
-    attrs = {
-        "python_interpreter": attr.label(
-            default = interpreter,
-            allow_single_file = True,
-            doc = "Python interpreter to use.",
-        ),
-        "requirements": attr.label(
-            default = "//:python-requirements.txt",
-            allow_single_file = True,
-            doc = "Python requirements file describing package dependencies.",
-        ),
-        "quiet": attr.bool(
-            default = True,
-            doc = "If True, suppress printing stdout/stderr to the terminal.",
-        ),
-        "timeout": attr.int(
-            default = 300,
-            doc = "Timeout (in seconds) on the rule's execution duration.",
-        ),
-        "_gen_requirements_script": attr.label(
-            default = "//third_party/python:gen_requirements.sh",
```

Comment:
If we no longer need this hack of pre-processing the python-requirements.txt file to support airgapped builds, then I think we can also delete the gen_requirements.sh script too. @jwnrt

Reply:
Ah yes, thanks, you're right
https://github.com/lowRISC/opentitan/pull/24591